packages/client/test/net/protocol/snapprotocol.spec.ts

```diff
@@ -82,7 +85,88 @@ tape('[SnapProtocol]', (t) => {
     t.end()
   })
 
-  t.test('verify that GetStorageRanges handler encodes/decodes correctly', (t) => {
+  t.test('AccountRange should encode/decode correctly', (t) => {
+    const config = new Config({ transports: [] })
+    const chain = new Chain({ config })
+    const p = new SnapProtocol({ config, chain })
+    /* eslint-disable @typescript-eslint/no-use-before-define */
+    const data = RLP.decode(Buffer.from(contractAccountRangeRLP, 'hex')) as unknown
+    const { reqId, accounts, proof } = p.decode(
+      p.messages.filter((message) => message.name === 'AccountRange')[0],
+      data
+    )
+    t.ok(reqId === BigInt(1), 'reqId should be 1')
+    t.ok(accounts.length === 2, 'accounts should be 2')
+    t.ok(proof.length === 7, 'proof nodes should be 7')
+
+    const firstAccount = accounts[0].body
+    const secondAccount = accounts[1].body
+
+    t.ok(firstAccount[2].length === 0, 'Slim format storageRoot for first account')
+    t.ok(firstAccount[3].length === 0, 'Slim format codehash for first account')
+    t.ok(
+      secondAccount[2].toString('hex') ===
+        '3dc6d3cfdc6210b8591ea852961d880821298c7891dea399e02d87550af9d40e',
+      'storageHash of the second account'
+    )
+    t.ok(
+      secondAccount[3].toString('hex') ===
+        'e68fe0bb7c4a483affd0f19cc2b989105242bd6b256c6de3afd738f8acd80c66',
+      'codeHash of the second account'
+    )
+    const payload = RLP.encode(
+      p.encode(p.messages.filter((message) => message.name === 'AccountRange')[0], {
+        reqId,
+        accounts,
+        proof,
+      })
+    )
+    t.ok(
+      contractAccountRangeRLP === Buffer.from(payload).toString('hex'),
+      'Re-encoded payload should match with original'
+    )
+    t.end()
+  })
+
+  t.test('AccountRange should verify a real sample', async (t) => {
+    const config = new Config({ transports: [] })
+    const chain = new Chain({ config })
+    const p = new SnapProtocol({ config, chain })
+
+    /* eslint-disable @typescript-eslint/no-use-before-define */
+    const reqData = RLP.decode(Buffer.from(getAccountRangeRLP, 'hex'))
+    const { root: stateRoot } = p.decode(
+      p.messages.filter((message) => message.name === 'GetAccountRange')[0],
+      reqData
+    )
+    // accountRangeRLP is the corresponding response to getAccountRangeRLP
+    const resData = RLP.decode(Buffer.from(accountRangeRLP, 'hex')) as unknown
+    const { accounts, proof } = p.decode(
+      p.messages.filter((message) => message.name === 'AccountRange')[0],
+      resData
+    )
+
+    const trie = new CheckpointTrie({ db: new LevelDB() })
+    try {
+      const accountRLP = await trie.verifyProof(
```

Comment:
If possible could this be changed to the `verifyRangeProof`? This should contain the logic which we should have to verify the entire proof. (Note: there could be bugs here, since I got those locally)

Reply:
we can replace with verifyRangeProof once the bugs are resolved and merged (change can be part of bug resolving PR itself)

Reply:
replaced with verifyRangeProof :+1: